### PR TITLE
add new test cases for CREATE and READ api

### LINKED ISSUE #6: Add the service to CREATE a wishlist
**As an** eCommerce website user
**I need** the wishlist creation service to create my own wishlist.
**So that** I could add a collection of products that I wish to buy.

**Assumptions:** 
* Vagrant file and Skeleton Flask is set up
* Data model for wishlist is defined

**Acceptance Criteria:**
```
Given - I have a create API
When I put a POST request on '/wishlists' 
Then a new record will be created in the database
And a status code "201 Created" will be returned 
And URL to get the wishlist will be returned in a Location header
```

### LINKED ISSUE #9: Add the service to READ a wishlist
**As a** User
**I need** the ability to read one of my wishlists
**So that** I can review the content of the wishlist, including a collection of products that I wish I had the money to purchase

**Assumptions:** 
* The wishlist ID should be passed in by the user
* A call to ``` GET  /wishlists/{id} ``` should show the content of the desired wishlist

**Acceptance Criteria:**

Given - I have a read API
When I call ``` GET /wishlists/{id} ```
I should see the wishlist that I want to review for the given id

### PR DESCRIPTION
**Pull request by yf1357**

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -123,7 +123,26 @@ def test_get_wishlist_not_found(self):
...
-   
+	def test_create_wishlist_with_missing_args(self):
+        test_wishlist = {
+            "name":"wishlist1",
+            "user_id":1
+            }
+        resp = self.app.post(
+            "/wishlists", json=test_wishlist, content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
```
Comment (abhinavGupta16) [inline tests/test_service.py:134]: We'll need to check if the message in the response is accurate. Same for the rest. This test case should **ideally fail** as our APIs promise a json response and the response here is HTML

However, that would require merging #23 I believe

**Review by sqz0914 — APPROVED** [review]

**Review by xiafanzeng — APPROVED** [review]

**Comment by abhinavGupta16** [comment]
@xiafanzeng @sqz0914 @yf1357 - The test cases are not complete. I wonder why we approved and merged it?

File: tests/test_service.py (hunk 1 above)
Comment (yf1357) [reply tests/test_service.py:134]: Sorry, I didn't get your point. The test case runs successfully and is only a supplement for error handling. If you are talking about response including content checking, it is included in last PR.
image

File: tests/test_service.py [hunk 2]
```diff
@@ -123,7 +123,26 @@ def test_get_wishlist_not_found(self):
         resp = self.app.get("/wishlists/0")
         self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
 
-   
+	def test_create_wishlist_with_missing_args(self):
```
Comment (abhinavGupta16) [inline tests/test_service.py:126]: @yf1357 - There is a tab error here, the master branch is breaking. Please check.

image

File: tests/test_service.py (hunk 1 above)
Comment (abhinavGupta16) [reply tests/test_service.py:134]: The response does give a 404, but as an HTML response. We are to only return json responses. If you test for the error message, the test case will fail.

Irrespectively, the test case should include the expected error message in the response with the error.

**Pull request merged**